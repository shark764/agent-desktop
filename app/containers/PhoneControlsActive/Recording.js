/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import Toggle from 'react-toggle';
import 'assets/css/react-toggle-style.css';

import messages from './messages';

const styles = {
  recordingContainerToolbar: {
    position: 'absolute',
    right: '20px',
    fontSize: '12px',
    fontWeight: 'bold',
  },
  recordingContainer: {
    padding: '0 12px 14px',
  },
  recordingText: {
    fontSize: '14px',
  },
  recordingToggleContainer: {
    float: 'right',
    verticalAlign: 'top',
    height: 20,
    marginTop: 2,
  },
  toggleRecordingLabel: {
    fontSize: '12px',
    verticalAlign: 'top',
    lineHeight: '21px',
    marginRight: '3px',
  },
};

export default class Recording extends React.PureComponent {
  setRecording = () => {
    if (this.props.isRecording) {
      CxEngage.interactions.voice.stopRecording({
        interactionId: this.props.interactionId,
      });
    } else {
      CxEngage.interactions.voice.startRecording({
        interactionId: this.props.interactionId,
      });
    }
  };

  render() {
    if (
      this.props.agentRecordingEnabled &&
      !this.props.preventAgentRecordingUpdate
    ) {
      const toggle = (
        <Toggle
          id="toggleRecording"
          icons={false}
          onChange={this.setRecording}
          checked={this.props.isRecording}
        />
      );
      if (this.context.toolbarMode) {
        return (
          <div id="recordingContainer" style={styles.recordingContainerToolbar}>
            <div>
              <FormattedMessage {...messages.rec} />
            </div>
            {toggle}
          </div>
        );
      } else {
        return (
          <div id="recordingContainer" style={styles.recordingContainer}>
            <span style={styles.recordingText}>
              <FormattedMessage {...messages.recording} />
            </span>
            <span style={styles.recordingToggleContainer}>
              <label
                htmlFor="toggleRecording"
                style={styles.toggleRecordingLabel}
              >
                {this.props.isRecording ? (
                  <FormattedMessage {...messages.on} />
                ) : (
                  <FormattedMessage {...messages.off} />
                )}
              </label>
              {toggle}
            </span>
          </div>
        );
      }
    }
    return null;
  }
}

Recording.propTypes = {
  interactionId: PropTypes.string.isRequired,
  isRecording: PropTypes.bool.isRequired,
  agentRecordingEnabled: PropTypes.bool.isRequired,
  preventAgentRecordingUpdate: PropTypes.bool.isRequired,
};

Recording.contextTypes = {
  toolbarMode: PropTypes.bool,
};
