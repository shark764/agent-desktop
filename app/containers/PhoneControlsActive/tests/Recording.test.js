/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';

import Recording from '../Recording';

describe('<Recording />', () => {
  describe('with agentRecordingEnabled and false preventAgentRecordingUpdate', () => {
    describe('not in toolbarMode', () => {
      const rendered = shallow(
        <Recording
          interactionId="mock-interaction-id"
          isRecording={false}
          agentRecordingEnabled
          preventAgentRecordingUpdate={false}
        />
      );
      it('renders correctly', () => {
        expect(rendered).toMatchSnapshot();
      });
    });
    describe('in toolbarMode', () => {
      const rendered = shallow(
        <Recording
          interactionId="mock-interaction-id"
          isRecording={false}
          agentRecordingEnabled
          preventAgentRecordingUpdate={false}
        />,
        {
          context: { toolbarMode: true },
        }
      );
      it('renders correctly', () => {
        expect(rendered).toMatchSnapshot();
      });
    });

    describe('with isRecording', () => {
      beforeEach(() => {
        global.CxEngage = {
          interactions: {
            voice: {
              stopRecording: jest.fn(),
            },
          },
        };
      });
      const rendered = shallow(
        <Recording
          interactionId="mock-interaction-id"
          isRecording
          agentRecordingEnabled
          preventAgentRecordingUpdate={false}
        />
      );
      it('calls CxEngage.interactions.voice.stopRecording with the interactionId', () => {
        rendered.find('#toggleRecording').simulate('change');
        expect(
          global.CxEngage.interactions.voice.stopRecording.mock.calls.length
        ).toBe(1);
        expect(
          global.CxEngage.interactions.voice.stopRecording.mock.calls[0][0]
        ).toEqual({ interactionId: 'mock-interaction-id' });
      });
    });
    describe('with false isRecording', () => {
      beforeEach(() => {
        global.CxEngage = {
          interactions: {
            voice: {
              startRecording: jest.fn(),
            },
          },
        };
      });
      const rendered = shallow(
        <Recording
          interactionId="mock-interaction-id"
          isRecording={false}
          agentRecordingEnabled
          preventAgentRecordingUpdate={false}
        />
      );
      it('calls CxEngage.interactions.voice.startRecording with the interactionId', () => {
        rendered.find('#toggleRecording').simulate('change');
        expect(
          global.CxEngage.interactions.voice.startRecording.mock.calls.length
        ).toBe(1);
        expect(
          global.CxEngage.interactions.voice.startRecording.mock.calls[0][0]
        ).toEqual({ interactionId: 'mock-interaction-id' });
      });
    });
  });

  describe('with agentRecordingEnabled false', () => {
    const rendered = shallow(
      <Recording
        interactionId="mock-interaction-id"
        isRecording={false}
        agentRecordingEnabled={false}
        preventAgentRecordingUpdate={false}
      />
    );
    it('renders nothing', () => {
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('with preventAgentRecordingUpdate', () => {
    const rendered = shallow(
      <Recording
        interactionId="mock-interaction-id"
        isRecording={false}
        agentRecordingEnabled
        preventAgentRecordingUpdate
      />
    );
    it('renders nothing', () => {
      expect(rendered).toMatchSnapshot();
    });
  });
});
