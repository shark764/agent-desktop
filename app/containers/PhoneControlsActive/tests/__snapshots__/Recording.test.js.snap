// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Recording /> with agentRecordingEnabled and false preventAgentRecordingUpdate in toolbarMode renders correctly 1`] = `
<div
  id="recordingContainer"
  style={
    Object {
      "fontSize": "12px",
      "fontWeight": "bold",
      "position": "absolute",
      "right": "20px",
    }
  }
>
  <div>
    <FormattedMessage
      defaultMessage="REC"
      id="app.containers.PhoneControls.rec"
      values={Object {}}
    />
  </div>
  <Toggle
    checked={false}
    icons={false}
    id="toggleRecording"
    onChange={[Function]}
  />
</div>
`;

exports[`<Recording /> with agentRecordingEnabled and false preventAgentRecordingUpdate not in toolbarMode renders correctly 1`] = `
<div
  id="recordingContainer"
  style={
    Object {
      "padding": "0 12px 14px",
    }
  }
>
  <span
    style={
      Object {
        "fontSize": "14px",
      }
    }
  >
    <FormattedMessage
      defaultMessage="Recording"
      id="app.containers.PhoneControls.recording"
      values={Object {}}
    />
  </span>
  <span
    style={
      Object {
        "float": "right",
        "height": 20,
        "marginTop": 2,
        "verticalAlign": "top",
      }
    }
  >
    <label
      htmlFor="toggleRecording"
      style={
        Object {
          "fontSize": "12px",
          "lineHeight": "21px",
          "marginRight": "3px",
          "verticalAlign": "top",
        }
      }
    >
      <FormattedMessage
        defaultMessage="OFF"
        id="app.containers.PhoneControls.off"
        values={Object {}}
      />
    </label>
    <Toggle
      checked={false}
      icons={false}
      id="toggleRecording"
      onChange={[Function]}
    />
  </span>
</div>
`;

exports[`<Recording /> with agentRecordingEnabled false renders nothing 1`] = `""`;

exports[`<Recording /> with preventAgentRecordingUpdate renders nothing 1`] = `""`;
